src/lazy/raw_stream_item.rs

```diff
@@ -116,12 +127,17 @@ impl<M: Copy + Debug, V: Copy + Debug, E: Copy + Debug> RawStreamItem<M, V, E> {
 /// an `EndOfStream(EndPosition)` variant) to also implement them.
 #[derive(Debug, Copy, Clone)]
 pub struct EndPosition {
+    encoding: IonEncoding,
     position: usize,
 }
 
 impl EndPosition {
-    pub(crate) fn new(position: usize) -> Self {
-        Self { position }
+    pub(crate) fn new(encoding: IonEncoding, position: usize) -> Self {
```

Comment:
🗺️ While it seems a bit silly for the end of the stream to have an 'encoding' despite being a zero-length entity, this enables the `LazyRawStreamItem<AnyEncoding>` to _always_ report its encoding.